src/app/rpmostree-builtin-status.c

```diff
@@ -48,24 +53,42 @@ static GOptionEntry option_entries[] = {
   { NULL }
 };
 
+static const char *bold_prefix = "";
+static const char *bold_suffix = "";
+static const char *red_prefix = "";
+static const char *red_suffix = "";
+
+static gboolean
+is_tty ()
+{
+  static gboolean is_tty;
+  static gsize initialized;
+  if (g_once_init_enter (&initialized))
+    {
+      is_tty = (isatty (STDOUT_FILENO) != 0);
+      g_once_init_leave (&initialized, 1);
+    }
+  return is_tty;
+}
+
 static void
-printpad (char c, guint n)
+init_escape_chars ()
 {
-  for (guint i = 0; i < n; i++)
-    putc (c, stdout);
+  if (is_tty ())
+    {
+      bold_prefix = "\x1b[1m";
+      bold_suffix = "\x1b[0m";
+      red_prefix = "\x1b[31m";
+      red_suffix = "\x1b[22m";
```

Comment:
It might make for slightly uglier code but maybe let's avoid the global variables and do e.g.:

```
static const char *
get_bold_prefix (void)
{
  if (is_tty ())
     return "\x1b[1m";
  return "";
}
```
etc?

Reply:
And while we're here hoist it to `main.c` or so?

Reply:
And split this into a prep patch?

Reply:
Good idea! This is done in https://github.com/projectatomic/rpm-ostree/pull/1196.